src/nodes.ts

```diff
@@ -1525,6 +1541,25 @@ export class NewOp extends Node {
   }
 }
 
+export class SoakedNewOp extends Node {
+  readonly ctor: Node;
+  readonly arguments: Array<Node>;
+
+  constructor(
+    line: number,
+    column: number,
+    start: number,
+    end: number,
+    raw: string,
+    ctor: Node,
+    args: Array<Node>
```

Comment:
I'm trying to use the shorthand syntax here:

`readonly arts: Array<Node>`

That makes it so you don't have to declare the property separately or explicitly assign it.